### PR TITLE
[#175798769] Start session guest user

### PR DESCRIPTION
**Pull request by GiovanniMancini**
This PR proposes to use the autogenerated PM Client to start a guest payment session with the Payment Manager. The client is integrated in front-end code. Moreover, integration with browser and PM is tested in a local environment, based on a dockerized version of PM.

#### List of Changes
- Autogeneration of PM client from customized `openapi` spec, based on the one exposed by PM `api-docs`
- Integration of autogerated PM client in frontend code
- Test of `start-session` in browser environment calling a dockerized version of PM
- Test of retryable fetch in browser environment calling a stub version of PM to emulate transient errors (429)
- Downgrade of `tsc` target to support older browsers
- Got rid of PM Client imported from IO APP
- Removed endpoint `/v3/psps/seller/list` from `api.yml` as one of the parameter is typed as `array` but such tagging is not supported by `swagger 2.0` so it doesn't validate

#### Motivation and Context

This PR solves the problem of integrating PM client in frontend code to start the guest payment session with the Payment Manager. Moreover it proposes a way to test in integration with browser and a dockerized version of PM

#### How Has This Been Tested?
I've included two integration test suites.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -149,14 +154,29 @@ document.addEventListener('DOMContentLoaded', () => {
...
+          },
+        },
+      });
 
-    creditcardformInputs?.forEach(el => {
-      sessionStorage.setItem(el.getAttribute('name')?.trim() || '', el.value);
-    });
-    window.location.replace('check.html');
-  });
+      if (result.isRight()) {
```
Comment (infantesimone) [inline src/index.ts:171]: According to [this](https://pagopa.github.io/io-docs/io-handbook/development-styleguide), you can use _fold_ instead _isRight_.
Something like that:

```ts
      (
        await pmClient.startSessionUsingPOST({
          startSessionRequest: {
            data: {
              email: 'pippo@pluto.com',
              fiscalCode: 'HBBJUU78U89R556T',
              idPayment: '12345',
            },
          },
        })
      ).fold(
        () => console.log('some error'),  // to be replaced with logic to handle failures
        () => {
          creditcardformInputs?.forEach(el => {
            sessionStorage.setItem(el.getAttribute('name')?.trim() || '', el.value);
          });
          window.location.replace('check.html');
        },
      );
```

File: tsconfig.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "compilerOptions": {
 
-    "target": "ES2019",
+    "target": "ES5",
     "module": "commonjs",
     "lib": ["ES6", "DOM"],
```
Comment (pasqualespica) [inline tsconfig.json:6]: Why this change ? Why do you use `target ES5` and then `lib ES6` ?
For compatibility with old browsers but to be able to use `Promise` for example?

**Review by pasqualespica — COMMENTED** [review]
can you better explain PR description, why this is removed?
Removed endpoint `/v3/psps/seller/list from api.yml`, `swagger 2.0` doesn't support arrays ?

File: package.json [hunk 3]
```diff
@@ -5,23 +5,24 @@
   "license": "MIT",
   "pagopa_api_walletv2": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/bonus/specs/bpd/pm/walletv2.json",
   "pagopa_api": "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/pagopa/pm/spec.json",
-  "generate:pagopa-api-definitions": "rimraf definitions/pagopa && mkdir -p definitions/pagopa && gen-api-models --api-spec $npm_package_pagopa_api --out-dir ./definitions/pagopa --no-strict --request-types --response-decoders --default-error-type undefined",
+  "pagopa_api_live": "api.yml",
```
[the remark this answers is not in the document]
Comment (GiovanniMancini) [reply package.json:8]: @infantesimone Thank you for pointing out. _live_ is just to remark a difference from the old specs, which are dead. Maybe I should have written _**a**live_ :-) . I'll remove old specs, old client and fix the name

File: package.json [hunk 4]
```diff
@@ -5,23 +5,24 @@
...
     "generate:pagopa-api-definitions": "rimraf generated/definitions/pagopa && mkdir -p generated/definitions/pagopa && gen-api-models --api-spec $npm_package_pagopa_api --out-dir ./generated/definitions/pagopa --no-strict --request-types --response-decoders --default-error-type undefined",
     "generate:pagopa-api-walletv2": "rimraf generated/definitions/pagopa/walletv2 && mkdir -p generated/definitions/pagopa/walletv2  && gen-api-models --api-spec $npm_package_pagopa_api_walletv2 --out-dir ./generated/definitions/pagopa/walletv2 --no-strict --request-types --response-decoders",
+    "generate:pagopa-api-live-client": "rimraf generated/definitions/pagopa_live && mkdir -p generated/definitions/pagopa_live && gen-api-models --api-spec $npm_package_pagopa_api_live --out-dir ./generated/definitions/pagopa_live --no-strict --request-types --response-decoders --client",
     "lint": "eslint . -c .eslintrc.js --ext .ts,.tsx",
     "lint-autofix": "eslint . -c .eslintrc.js --ext .ts,.tsx --fix",
-    "pretest:integration": "yarn build:test-cases",
-    "pretest:coverage-int": "yarn build:test-cases",
+    "pretest:integration": "yarn build && yarn build:test-cases",
```
[the remark this answers is not in the document]
Comment (GiovanniMancini) [reply package.json:16]: @infantesimone As you may know, the target _pre_ are automatically executed before the similar target without _pre_. In Azure pipelines tests are executed using the target `test:coverage`, which runs `test:coverage-int` and in turn `pretest:coverage-int`. When I develop in a local environment I prefer to not compute the coverage whenever I run the tests, so I needed a specific _pre_ target for the script `yarn test:integration`

File: src/__integrations__/devServer.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import express from 'express';
```
[the remark this answers is not in the document]
Comment (GiovanniMancini) [reply src/__integrations__/devServer.ts:1]: @infantesimone the `devServer` is almost useless, since is serves static content and it can be substituted by a one-liner. Maybe I could remove it

File: src/__integrations__/pmBad.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+// import { debug as cdebug } from 'console';
+
+import { Application, Router } from 'express';
```
[the remark this answers is not in the document]
Comment (GiovanniMancini) [reply src/__integrations__/pmBad.ts:3]: @infantesimone Ok, I'll merge the two servers for maintainability and readability

File: src/index.ts [hunk 7]
```diff
@@ -1,11 +1,17 @@
...
+import { createClient } from '../generated/definitions/pagopa_live/client';
 import { getUrlParameter } from './js/urlUtilities';
 import { setTranslateBtns } from './js/translateui';
 import { modalWindows } from './js/modals';
 import { userSession } from './js/sessiondata';
 
 // eslint-disable-next-line sonarjs/cognitive-complexity
 document.addEventListener('DOMContentLoaded', () => {
+  const pmClient = createClient({
+    baseUrl: 'http://localhost:8080',
```
[the remark this answers is not in the document]
Comment (GiovanniMancini) [reply src/index.ts:11]: @infantesimone Ok, I'll fix as you suggest

File: src/index.ts (hunk 1 above)
Comment (GiovanniMancini) [reply src/index.ts:171]: @infantesimone Thank you for the snippet! I'll use it

**Comment by GiovanniMancini** [comment]
> can you better explain PR description, why this is removed?
> Removed endpoint `/v3/psps/seller/list from api.yml`, `swagger 2.0` doesn't support arrays ?

File: tsconfig.json (hunk 2 above)
Comment (GiovanniMancini) [reply tsconfig.json:6]: @pasqualespica I found that setting target `ES5` and lib `ES6` supports `async`/`await` syntax for promises and also supports old browsers. I couldn't make things work with different choices

**Comment by GiovanniMancini** [comment]
@pasqualespica  OK, I'll improve the description
> can you better explain PR description, why this is removed?
> Removed endpoint `/v3/psps/seller/list from api.yml`, `swagger 2.0` doesn't support arrays ?

File: src/index.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (infantesimone) [reply src/index.ts:11]: Maybe `dotenv` dependency is not enough, how are variables loaded? 
You need a module for this, maybe you can follow the strategy of [io-pay-portal](https://github.com/pagopa/io-pay-portal/blob/main/io-pay-portal-fe/env.sh) or [io-developer-portal](https://github.com/pagopa/io-developer-portal-frontend/blob/master/env.sh).

[the remark this answers is not in the document]
Comment (infantesimone) [reply src/index.ts:11]: To explore in another PR

**Review by infantesimone — APPROVED** [review]
